packages/fastify-renderer/package.json

```diff
@@ -68,8 +68,10 @@
     "path-to-regexp": "^6.2.1",
     "sanitize-filename": "^1.6.3",
     "stream-template": "^0.0.10",
+    "ts-node": "^10.9.1",
     "vite": "^2.9.15",
-    "wouter": "^2.7.5"
+    "wouter": "^2.7.5",
+    "resource-pooler": "^0.1.0"
```

Comment:
Used for Worker pooling: https://github.com/vezaynk/resource-pooler

Reply:
I know that piscina had a lot of work put into it to make it fast -- worth doing a quick benchmark to make sure that this implementation isn't far far worse?

Reply:
Code is faster when it does less work 😉 

It's basically an async queue of Workers that are being re-used. It all runs in O(1) time.

Reply:
Right! I took a quick look and spotted a couple issues:
 - it doesn't look like failures in the worker or in the dispose function are handled, and without that the internal tracking state will get corrupted. I think that those should be expected, especially as things like react server components which might make DB calls and whatnot come down the pipes
 - either here or other, can the queue be initialized to the number of CPUs instead of a hardcoded 8? since this worker is purely CPU bound I don't think having more workers will increase throughput, it will just increase context switching
 - is it worth having a dequeue timeout / render timeout in case something goes wrong

Reply:
Just taking a brief look at the code for the pooler I see a couple issues:
 - it doesn't look like failing tasks will be covered, if the run function errors

Reply:
> it doesn't look like failures in the worker or in the dispose function are handled

That's true. I'll revisit this. 

With that said, failures in the rendering wouldn't make `resource-pooler` misbehave. The Worker will just experience a promise rejection.

Given that I can't send an error as-is from a worker back to the main thread, how should SSR rendering errors be reported?

I'm thinking of merely detecting that _something_ went wrong on the main thread and sending back a 500, while the worker logs more detail with `console.error`.

> either here or other, can the queue be initialized to the number of CPUs instead of a hardcoded 8?

Yea you're right. I'll do it in here, because `resource-pooler` could feasibly be used outside of Nodejs/multi-threading contexts.

> is it worth having a dequeue timeout / render timeout in case something goes wrong

What do you think? It's easy to add but I can't imagine a situation that would be remediated by waiting a bit.

Reply:
I improved error-handling on resource-pooler.

> Given that I can't send an error as-is from a worker back to the main thread, how should SSR rendering errors be reported?

So this was untrue. Errors _can_ be forwarded as-is. I refactored the passthrough stream to be more type-safe as well as forwarding errors.